### PR TITLE
476 - Study - Middle layer to address XML applications

### PR DESCRIPTION
**Pull request by florentmaitre**

### CODE REVIEW

File: app/src/main/java/com/orange/ods/app/ui/about/AboutNavGraph.kt [hunk 1]
```diff
@@ -15,13 +15,16 @@ import androidx.navigation.NavType
...
 import com.orange.ods.app.ui.MainDestinations
+import com.orange.ods.app.ui.MainTopAppBarState
 
 fun NavGraphBuilder.addAboutGraph() {
     composable(
         "${MainDestinations.AboutItemDetailRoute}/{${MainDestinations.AboutItemIdKey}}",
         arguments = listOf(navArgument(MainDestinations.AboutItemIdKey) { type = NavType.LongType })
     ) { backStackEntry ->
+        LocalMainTopAppBarManager.current.updateTopAppBar(MainTopAppBarState.DefaultConfiguration)
         LocalMainTabsManager.current.clearTopAppBarTabs()
```
Comment (paulinea) [inline app/src/main/java/com/orange/ods/app/ui/about/AboutNavGraph.kt:28]: We can maybe create a `reset` function to apply default configuration and clear the tabs cause it is done many times. What do you think?

File: app/src/main/java/com/orange/ods/app/ui/components/ComponentsNavGraph.kt [hunk 2]
```diff
@@ -44,6 +46,15 @@ fun NavGraphBuilder.addComponentsGraph(navigateToElement: (String, Long?, NavBac
     ) { from ->
         val arguments = requireNotNull(from.arguments)
         val variantId = arguments.getLong(MainDestinations.ComponentVariantIdKey)
+        val topAppBarConfiguration = MainTopAppBarState.DefaultConfiguration.newBuilder()
```
Comment (paulinea) [inline app/src/main/java/com/orange/ods/app/ui/components/ComponentsNavGraph.kt:49]: As discussed, move framework switch in the customization bottom sheet

**Review by paulinea — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/orange/ods/app/ui/components/banners/ComponentBanners.kt [hunk 3]
```diff
@@ -72,35 +77,61 @@ fun ComponentBanners() {
...
+                            ) {
+                                if (hasTwoTextLines) TechnicalText(text = " message = \"${recipe.description}\"")
+                                else TechnicalText(text = " message = \"${recipe.title}\"")
+                                if (hasIcon) TechnicalText(text = " image = painterResource(id = R.drawable.image)")
+                                TechnicalText(text = " button1Text = \"${stringResource(id = R.string.component_banner_dismiss)}\"")
+                                if (hasButton2) TechnicalText(text = " button2Text = \"${stringResource(id = R.string.component_banner_detail)}\"")
+                            }
+                        }
+                    },
+                    xml = {
```
Comment (paulinea) [inline app/src/main/java/com/orange/ods/app/ui/components/banners/ComponentBanners.kt:115]: To be complete, we also have to display code implementation for XML and to add XML components documentation. Maybe create another issue for that?

Comment (florentmaitre) [reply app/src/main/java/com/orange/ods/app/ui/components/banners/ComponentBanners.kt:115]: Yes, it's probably a lot of work, so I thought we could do this in another PR :+1:

File: app/src/main/java/com/orange/ods/app/ui/about/AboutNavGraph.kt (hunk 1 above)
Comment (paulinea) [reply app/src/main/java/com/orange/ods/app/ui/about/AboutNavGraph.kt]: I'm not really fan of ` ResetTopAppBar()` composable. I suggest another thing I tried on my side (I stash my work, it works). I added the tabs management in the top app bar cause tabs are for me a part of this bar. I'll show you when you'll be back.

File: app/src/main/java/com/orange/ods/app/ui/components/banners/ComponentBanners.kt (hunk 3 above)
Comment (paulinea) [reply app/src/main/java/com/orange/ods/app/ui/components/banners/ComponentBanners.kt:126]: I'm going to create a new issue for that

**Pull request merged**